src/rtc_base/platform_thread_types.h

```diff
@@ -38,6 +40,9 @@ typedef DWORD PlatformThreadRef;
 #elif defined(WEBRTC_FUCHSIA)
 typedef zx_handle_t PlatformThreadId;
 typedef zx_handle_t PlatformThreadRef;
+#elif defined(WEBRTC_OPENBSD)
+typedef uint64_t PlatformThreadId;
```

Comment:
Why pid_t doesn't fit? :thinking:

Reply:
`pid_t` aka. `int` on OpenBSD/amd64 causes the following error:
```
platform_thread_types.cc:57:10: error: cast from pointer to smaller type 'rtc::PlatformThreadId' (aka 'int') loses information
  return reinterpret_cast<PlatformThreadId>(pthread_self());
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

Reply:
Shouldn't this be a warning? Maybe casting to a smaller type is intended here?

Reply:
Ah, looks like solaris uses has `typedef long pid_t;`. I think, the right fix is to replace pid_t with uintptr_t.

Reply:
> Ah, looks like solaris has `typedef long pid_t;`. I think, the right fix is to replace pid_t with uintptr_t.

Yes, `uintptr_t` works but requires an additional `#include <stdint.h>`.
`intptr_t` works on OpenBSD with the existing includes under `WEBRTC_POSIX` as it is defined in `<unistd.h>` instead.

I was hesitant to change that line as I cannot test on other (POSIX) platforms.

Reply:
Let's try intptr_t. If someone would complain, this can be reverted.